integrations/anthropic/src/haystack_integrations/components/generators/anthropic/chat/chat_generator.py

```diff
@@ -0,0 +1,281 @@
+import dataclasses
+from typing import Any, Callable, ClassVar, Dict, List, Optional, Union
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import ChatMessage, ChatRole, StreamingChunk
+from haystack.utils import Secret, deserialize_callable, deserialize_secrets_inplace, serialize_callable
+
+from anthropic import Anthropic, Stream
+from anthropic.types import (
+    ContentBlock,
+    ContentBlockDeltaEvent,
+    Message,
+    MessageDeltaEvent,
+    MessageStartEvent,
+    MessageStreamEvent,
+    TextDelta,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class AnthropicChatGenerator:
```

Comment:
`AnthropicChatGenerator` could inherit from `AnthropicGenerator` or the other around to reduce code duplication.
`_get_telemetry_data` and`from_dict`  are the same across both classes and so is ALLOWED_PARAMS.
`to_dict` differs as because of system prompt as you pointed out.

Reply:
If it weren't for the system prompt I would insist on using inheritance. Now with system prompt, there aren't many duplicated lines of code and in order to move fast I am okay with keeping the two classes as they are.